scheduler/src/cook/kubernetes/api.clj

```diff
@@ -1069,8 +1080,10 @@
     (.addTolerationsItem pod-spec toleration-for-deletion-candidate-of-autoscaler)
     (.addTolerationsItem pod-spec (toleration-for-pool cook-pool-taint-name cook-pool-taint-prefix pool-name))
     ; We need to make sure synthetic pods --- which don't have a hostname set --- have a node selector
-    ; to run only in nodes labelled with the appropriate cook pool
-    (when-not hostname (add-node-selector pod-spec cook-pool-label-name pool-name))
+    ; to run only in nodes labelled with the appropriate cook pool and only in un-tenured nodes.
+    (when-not hostname
+      (add-node-selector pod-spec cook-pool-label-name pool-name)
+      (.addTolerationsItem pod-spec toleration-tenured-node))
```

Comment:
I think this is making only synthetic pods tolerate the taint, instead of only real job pods